### PR TITLE
[4.0][a11y]Admin module logged button

### PR DESCRIPTION
**Pull request by chmst**
It is a a11y issue if links are used instead of buttons. 

### Summary of Changes
In admin module mod_logged, the button to log out a user is now a button, not a link. 
It is implemented as a form. 

### Testing Instructions
You must have at least one - or better several - user(s) logged in in frontend. 
Open the cpanel, the logged_in users are visible and each of them has a "logout" button. 
Have a look in the sourcecode and control this via axe and with keyboard and screenreader.

### Expected result
The buttons are accessible. When the admin activates one of the buttons, the respective user is logged out in the frontend.

### Actual result
The buttons is a link with role="button" and therefore not a11y. When the administrator activates one of the buttons, the respective user is logged out in the frontend.

### Documentation Changes Required
no

### CODE REVIEW

File: administrator/modules/mod_logged/tmpl/default.php [hunk 1]
```diff
@@ -29,39 +30,45 @@
...
+				<td>
+					<?php if ($user->client_id === null) : ?>
+						<?php // Don't display a client ?>
+					<?php elseif ($user->client_id) : ?>
+						<?php echo Text::_('JADMINISTRATION'); ?>
+					<?php else : ?>
+						<form action="<?php echo Route::_('index.php?option=com_login&task=logout'); ?>"  method="post" name="adminForm">
+							<?php echo Text::_('JSITE'); ?>
+							<span>
+								<button type="submit" class="mr-2 btn btn-danger btn-sm" role="button">
```
Comment (SharkyKZ) [inline administrator/modules/mod_logged/tmpl/default.php]:
```suggestion
								<button type="submit" class="mr-2 btn btn-danger btn-sm">
```

File: administrator/modules/mod_logged/tmpl/default.php [hunk 2]
```diff
@@ -29,39 +30,45 @@
...
+						<?php echo htmlspecialchars($user->name, ENT_QUOTES, 'UTF-8'); ?>
+					<?php endif; ?>
+				</th>
+				<td>
+					<?php if ($user->client_id === null) : ?>
+						<?php // Don't display a client ?>
+					<?php elseif ($user->client_id) : ?>
+						<?php echo Text::_('JADMINISTRATION'); ?>
+					<?php else : ?>
+						<form action="<?php echo Route::_('index.php?option=com_login&task=logout'); ?>"  method="post" name="adminForm">
```
Comment (SharkyKZ) [inline administrator/modules/mod_logged/tmpl/default.php]:
```suggestion
						<form action="<?php echo Route::_('index.php?option=com_login&task=logout'); ?>" method="post" name="adminForm">
```

**Comment by SharkyKZ** [comment]
Remove helper code related to `logoutLink`? Or use it in form action?

https://github.com/joomla/joomla-cms/blob/96bcefcdb31612742c185efd6df5c392de1e18dc/administrator/modules/mod_logged/Helper/LoggedHelper.php#L62-L67

**Comment by SharkyKZ** [comment]
Previous commit removed too much code. Only need to remove 2 lines containing `logoutLink`.

File: administrator/modules/mod_logged/tmpl/default.php [hunk 3]
```diff
@@ -29,39 +30,45 @@
...
+				</th>
+				<td>
+					<?php if ($user->client_id === null) : ?>
+						<?php // Don't display a client ?>
+					<?php elseif ($user->client_id) : ?>
+						<?php echo Text::_('JADMINISTRATION'); ?>
+					<?php else : ?>
+						<form action="<?php echo Route::_('index.php?option=com_login&task=logout'); ?>" method="post" name="adminForm">
+							<?php echo Text::_('JSITE'); ?>
+							<span>
```
Comment (Quy) [inline administrator/modules/mod_logged/tmpl/default.php]: Is wrapping in a `span` tag necessary?

Comment (chmst) [reply administrator/modules/mod_logged/tmpl/default.php]: It lets be the text "Site" and the button be in the same line. Otherwise we have two lines which looks even more ugly than now. Making it beautiful is another issue :)

Comment (Quy) [reply administrator/modules/mod_logged/tmpl/default.php]: Removed the `span` tag and button is still on the same line.

Comment (chmst) [reply administrator/modules/mod_logged/tmpl/default.php]: @Quy you are right :) I was on a small device.

File: administrator/modules/mod_logged/tmpl/default.php [hunk 4]
```diff
@@ -29,39 +30,43 @@
...
+						<?php echo htmlspecialchars($user->name, ENT_QUOTES, 'UTF-8'); ?>
+					<?php endif; ?>
+				</th>
+				<td>
+					<?php if ($user->client_id === null) : ?>
+						<?php // Don't display a client ?>
+					<?php elseif ($user->client_id) : ?>
+						<?php echo Text::_('JADMINISTRATION'); ?>
+					<?php else : ?>
+						<form action="<?php echo Route::_('index.php?option=com_login&task=logout'); ?>" method="post" name="adminForm">
```
Comment (Quy) [inline administrator/modules/mod_logged/tmpl/default.php]: Will it work to use the `get` method, use  `$user->logoutLink;` for the `action`, and undo the changes in `LoggedHelper.php`?

Comment (chmst) [reply administrator/modules/mod_logged/tmpl/default.php]: yes, finally it is the same function.

File: administrator/modules/mod_logged/tmpl/default.php [hunk 5]
```diff
@@ -29,39 +30,41 @@
...
+						<?php echo htmlspecialchars($user->name, ENT_QUOTES, 'UTF-8'); ?>
+					<?php endif; ?>
+				</th>
+				<td>
+					<?php if ($user->client_id === null) : ?>
+						<?php // Don't display a client ?>
+					<?php elseif ($user->client_id) : ?>
+						<?php echo Text::_('JADMINISTRATION'); ?>
+					<?php else : ?>
+						<form action="<?php echo Route::_($user->logoutLink); ?>" method="post" name="adminForm">
```
Comment (Quy) [inline administrator/modules/mod_logged/tmpl/default.php]:
```suggestion
						<form action="<?php echo $user->logoutLink; ?>" method="post" name="adminForm">
```

File: administrator/modules/mod_logged/tmpl/default.php [hunk 6]
```diff
@@ -11,6 +11,7 @@
 
 use Joomla\CMS\HTML\HTMLHelper;
 use Joomla\CMS\Language\Text;
+use Joomla\CMS\Router\Route;
```
Comment (Quy) [inline administrator/modules/mod_logged/tmpl/default.php]:
```suggestion
```

File: administrator/modules/mod_logged/tmpl/default.php (hunk 5 above)
Comment (Quy) [reply administrator/modules/mod_logged/tmpl/default.php]: Route is already handled in `LoggedHelper.php`.

**Comment by zwiastunsw** [comment]
I have tested this item :white_check_mark: successfully on a1a79119af27496b015d59bc5ddc1ac21ed83084<br /><br />Accessibility: OK
In my opinion, there is no need for an X sign. Unnecessary piece of code. It does not bring any new information.
PS. Shouldn't there be a "Force logout" label?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23277">issues.joomla.org/tracker/joomla-cms/23277</a>.</sub>

**Comment by chmst** [comment]
@zwiastunsw could you repeat the test? 
I agree with your proposal to remove the icon,.
But at the moment I do not want to add another Language Key to the system.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23277">issues.joomla.org/tracker/joomla-cms/23277</a>.</sub>

**Comment by fancyFranci** [comment]
I have tested this item :white_check_mark: successfully on 9abdc056e34153860abfd3475ba32eb9216516e8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23277">issues.joomla.org/tracker/joomla-cms/23277</a>.</sub>

**Comment by rdeutz** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23277">issues.joomla.org/tracker/joomla-cms/23277</a>.</sub>
